packages/server/lib/socket-base.ts

```diff
@@ -404,6 +405,12 @@ export class SocketBase {
                 runState = args[0]
 
                 return null
+              case 'preserve:run:actions':
```

Comment:
maybe we can add some tests in the future the exercise these two actions in an e2e test (but functions more as an integration like test) by stubbing the backend , similar to `files.cy.ts`